calico/reference/resources/felixconfig.md

```diff
@@ -132,12 +132,19 @@ policy is always accelerated, using the best available BPF technology.
 | protocol | The protocol match   | tcp, udp, sctp                       | string |
 | net      | The CIDR match       | any valid CIDR (e.g. 192.168.0.0/16) | string |
 
-#### RouteTableRange
+#### RouteTableRanges
+`RouteTableRanges` is a list of `RouteTableRange` objects:
+
+| Field    | Description          | Schema |
+|----------|----------------------|--------|
+| min      | Minimum index to use | int    |
+| max      | Maximum index to use | int    |
+
+Each `RouteTableRange` designates a set of routing tables available to Calico. Modern Linux kernels can theorectically support more than the documented 252 additional tables. However, allocating more than the default number of tables is not as simple as increasing the max table index, since the increased range would encompass the reserved tables, and could lead to Calico clashing with the kernel.
```

Comment:
Can live with this approach (we've got lots of config parameters that are similarly handled), but (with experience gained from all those parameters...) I think it'd be better for the user if you just automatically excluded the reserved Linux tables.  I.e. let the user specify `10-10000` and just discard the `25x` tables with an INFO log `"Route table range contains linux reserved range, excluding 253-255."`.  

Then I'd include a note in the docs that Felix automatically excludes `0 and 253-255` since they are reserved by Linux to let knowledgeable users know so they don't freak out.   Rule of thumb: if the explanation in the docs can be shorter/easier, it might be a good idea.

Reply:
That does sound like a favourable solution in fairness! I think the main reason I opted for this approach is that different provisioners will have different reserved tables - E.g, in the case of AWS, as noted in the docs. I'm not hugely familiar with how much variance there is across provisioners, but maintaining a list of which ones reserve which tables gave me a hunch it may be better to go with the verbose option.
Totally makes sense what you're saying though, and my worries may be overblown! :slightly_smiling_face:

Reply:
I was only thinking of ignoring the Linux-reserved tables.  Will still need config for other provisioners.